### ISSUE TITLE
Lab 5- Part 2 - Q2

### ISSUE THREAD

**Issue by AhmedRashwanASU** [DS4PS/cpp-526-spr-2021#13]
while running the below code, I'm not getting the correct values of the hour's info, as below what I'm missing here? 

dat %>% 
  group_by(age,hour12) %>% 
  summarize(n = n()) %>%    
  mutate( n.age = sum(n) ,n.hour12 = sum(n) ) %>%
  mutate( prop= round( n / n.age, 2 ) ) %>% 
  mutate( prop= round( n / n.hour12, 2 ) ) %>%                
  filter(hour12 == "7 AM") 

![image](https://user-images.githubusercontent.com/77247268/107766383-d0071800-6d4c-11eb-94b3-209f5ad37fdc.png)

**Comment by jamisoncrawford** [comment]
Hi @AhmedRashwanASU - the first thing that comes to mind is that your second call to `mutate()` creates the variable `prop`, but your third call to `mutate()` overwrites that same variable with a different formula.  You should only need that once (and you are looking for the one with the proportion of `n.age`).

The one other step that your missing is to include a `group_by()` call in between your `summarize()` and first `mutate()`.  You will have to group by a single variable that will help you calculate counts and proportions for each age group.

In short, your order of piped operations should be as follows:

1. Dataset
2. `group_by()`
3. `summarize()`
4. `group_by()`
5. `mutate()`
6. `mutate()`
7. `filter()`

In practice, you can call `filter()` nearly any time, though it's best at the beginning or end, IMO.  Does this help?
